apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx

```diff
@@ -34,6 +39,104 @@ export default function CurriculumQuickAssign({updateSection, sectionCourse}) {
       .then(data => setCourseOfferings(data));
   }, []);
 
+  useEffect(() => {
+    // if (!updateSection) return;
+    if (!courseOfferings) return;
+    // const updateCourse = course => updateSection('course', course);
+    if (!isNewSection) {
+      // note that the desire is to come back and refactor this section to
+      // make it more efficient
+      const highData = {
+        ...courseOfferings[MARKETING_AUDIENCE.HIGH][curriculumTypes.course],
+        ...courseOfferings[MARKETING_AUDIENCE.HIGH][
+          curriculumTypes.standalone_unit
+        ],
+        ...courseOfferings[MARKETING_AUDIENCE.HIGH][curriculumTypes.module]
+      };
+      const middleData = {
+        ...courseOfferings[MARKETING_AUDIENCE.MIDDLE][curriculumTypes.course],
+        ...courseOfferings[MARKETING_AUDIENCE.MIDDLE][
+          curriculumTypes.standalone_unit
+        ],
+        ...courseOfferings[MARKETING_AUDIENCE.MIDDLE][curriculumTypes.module]
+      };
+      const elementaryData = {
+        ...courseOfferings[MARKETING_AUDIENCE.ELEMENTARY][
+          curriculumTypes.course
+        ],
+        ...courseOfferings[MARKETING_AUDIENCE.ELEMENTARY][
+          curriculumTypes.module
+        ]
+      };
+      const hocData = {...courseOfferings[MARKETING_AUDIENCE.HOC]};
+
+      // Note to self: Needed to move this inside the useEffect or make a custom hook
+      const updateSectionCourseForExisitngSections = course => {
+        const courseVersions = {};
+        // The structure of cv is an array with the first item an id and the second
+        // item an object of everything. See 'CourseOfferingsTestData' for examples
+        course.course_versions.map(cv => {
+          courseVersions[cv[1].id] = cv[1];
+        });
+        const courseVersionId = sectionCourse.versionId;
+        // Determine if it is a stand alone unit
+        const courseVersion = courseVersions[courseVersionId];
+        const isStandaloneUnit = Object.keys(courseVersion.units).length < 2;
+
+        let targetUnit;
+
+        if (isStandaloneUnit) {
+          targetUnit = Object.values(courseVersion.units)[0];
+        } else if (sectionCourse.unitId) {
+          targetUnit = courseVersion.units[sectionCourse.unitId];
+        }
+
+        const updateSectionData = {
+          displayName: course.display_name,
+          courseOfferingId: course.id,
+          versionId: courseVersionId,
+          unitId: isStandaloneUnit ? null : sectionCourse.unitId,
+          hasLessonExtras: targetUnit?.lesson_extras_available,
+          hasTextToSpeech: targetUnit?.text_to_speech_enabled
+        };
+
+        updateSection('course', updateSectionData);
+      };
+
+      const determineSelectedCourseOffering = (startingData, audience) => {
```

Comment:
I think this helper function could be pulled outside of the useEffect so that the function a little shorter/easier to read. Up to you